tests/providers/azure/azure_test.py

```diff
@@ -13,3 +13,410 @@
+    disk = FAKE_INSTANCE.GetDisk('fake-disk-name')
+    self.assertEqual('fake-disk-name', disk.name)
+
+    with self.assertRaises(RuntimeError):
+      FAKE_INSTANCE.GetDisk('non-existent-disk-name')
+
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.ListDisks')
+  @mock.patch('azure.mgmt.compute.v2020_06_01.operations._virtual_machines_operations.VirtualMachinesOperations.get')
+  @typing.no_type_check
+  def testListDisks(self, mock_get_vm, mock_list_disk):
+    """Test that disks from an instance are correctly listed."""
+    mock_get_vm.return_value = mock.Mock(
+        storage_profile=mock.Mock(os_disk=MOCK_BOOT_DISK, data_disks=[]))
+
+    # MOCK_LIST_DISKS contains all 2 disks from the subscription
+    self.assertEqual(2, len(MOCK_LIST_DISKS))
+    mock_list_disk.return_value = MOCK_LIST_DISKS
+
+    # The instance is expected to have only a boot disk as we mocked the
+    # data_disks attribute with an empty list
+    instance_disks = FAKE_INSTANCE.ListDisks()
+    mock_list_disk.assert_called()
+    mock_list_disk.assert_called_with(
+        resource_group_name=FAKE_INSTANCE.resource_group_name)
+    self.assertEqual(1, len(instance_disks))
+    self.assertNotEqual(MOCK_LIST_DISKS, instance_disks)
+
+    # The instance is expected to have 2 disks as we mocked the
+    # data_disks attribute with a non-empty list
+    mock_get_vm.return_value = mock.Mock(
+        storage_profile=mock.Mock(
+            os_disk=MOCK_BOOT_DISK, data_disks=[MOCK_DISK]))
+    instance_disks = FAKE_INSTANCE.ListDisks()
+    mock_list_disk.assert_called()
+    mock_list_disk.assert_called_with(
+        resource_group_name=FAKE_INSTANCE.resource_group_name)
+    self.assertEqual(2, len(instance_disks))
+    self.assertEqual(MOCK_LIST_DISKS, instance_disks)
+
+
+class TestForensics(unittest.TestCase):
+  """Test Azure forensics file."""
+  # pylint: disable=line-too-long
+
+  @mock.patch('libcloudforensics.providers.azure.internal.common.GetCredentials')
+  @mock.patch('libcloudforensics.providers.azure.internal.compute.AZSnapshot.Delete')
+  @mock.patch('libcloudforensics.providers.azure.internal.compute.AZDisk.Snapshot')
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.GetDisk')
+  @mock.patch('libcloudforensics.providers.azure.internal.compute.AZVirtualMachine.GetBootDisk')
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.GetInstance')
+  @mock.patch('azure.mgmt.compute.v2020_05_01.operations._disks_operations.DisksOperations.create_or_update')
+  @typing.no_type_check
+  def testCreateDiskCopy1(self,
+                          mock_create_disk,
+                          mock_get_instance,
+                          mock_get_boot_disk,
+                          mock_get_disk,
+                          mock_snapshot,
+                          mock_snapshot_delete,
+                          mock_credentials):
+    """Test that a disk copy is correctly created."""
```

Comment:
I normally don't care too much about docstrings in tests, but since you have two very similar tests it would be nice to clarify the difference between both tests

Reply:
The difference is clarified right before the `CreateDiskCopy` call in a comment. Should we move this comment to the docstrings for similar tests?

Reply:
Ah I see it now! yeah, it might be better to put it in the docstring as this is also what shows up when the test fails.

Reply:
Ack, done